tests/testthat/test-run-default-load.R

```diff
@@ -0,0 +1,53 @@
+context("run-default-load")
+
+test_that("default load can be run", {
+  ## Set up a db with some people alread loaded
+  db_name <- "test.sqlite"
+  prepare_example_db(db_name, add_data = TRUE, add_job_table = TRUE)
+  on.exit(unlink(db_name))
+
+  ## Turn off reporting when running import so import tests do not print
+  ## to avoid cluttering up test output.
+  default_reporter <- testthat::default_reporter()
+  options(testthat.default_reporter = "silent")
+  on.exit(options(testthat.default_reporter = default_reporter), add = TRUE)
+
+  import <- dettl("example_default_load/", default_load = TRUE,
+                  db_name = "test")
+
+  ## Setup mock INSERT query for testing. This is needed because in real
+  ## impl we want to do an INSERT INTO RETURNING pattern but it is not possible
+  ## to INSERT and return id in SQLite. But SQLite does make a
+  ## "last_insert_rowid" function available so use this instead inside a mock
+  ## impl to update the table and get the ID of the inserted row back.
+  sql <- "INSERT INTO people
+          (name, age, height)
+          VALUES
+          ($1, $2, $3)"
+  con <- import$get_connection()
+  insert_func <- function(x) {
+    query <- DBI::dbSendQuery(con, sql, unname(x))
+    DBI::dbClearResult(query)
+    DBI::dbGetQuery(con, "SELECT last_insert_rowid()")[1, 1]
+  }
+  run_import(import, c("extract", "transform"))
+  data <- import$get_transformed_data()$people
+  mock_insert <- mockery::mock(insert_func(data[1, c("name", "age", "height")]),
+                               insert_func(data[2, c("name", "age", "height")]))
+
+  ## Run load with mock INSERT function
+  with_mock("dettl:::insert_data" = mock_insert, {
```

Comment:
Here I am mocking the code which does the actual insert as that uses an `INSERT INTO RETURNING` pattern which is not available in SQLite. This looked like the best way to work around it though I'm not super happy with it as it results in lots of code in the test and means I'm not testing the entire `get_default_load` function. I wonder if that is just something we have to accept as the test driver is using SQLite instead of Postgres so there will be some differences. 

I'm hoping the explanation lines 15-22 when I start writing the mocking is enough to make this clear too?

Reply:
This was following the examples from `mockery` documentation for `mock` and the "simple integration" section mockery [vignette](https://cran.r-project.org/web/packages/mockery/vignettes/mocks-and-testthat.html). Is there a mockery equivalent or `with_mock`?

I could avoid using the `mockery::mock` but I think that would make the mock function more complicated.